vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -845,7 +845,11 @@ def _collect_metrics_async(self, instance, query_specs):
                         continue
 
                     instance_name = result.id.instance or "none"
-                    value = self._transform_value(instance, result.id.counterId, result.value[0])
+                    # Get the most recent value that isn't negative
+                    valid_values = [v for v in result.value if v >= 0]
```

Comment:
Are you saying that there is not a single collected metric that can be negative ?

Reply:
Yes,
According to our `metadata.csv` units, this are all the units we support `['command', 'command/second', 'kibibyte', 'kibibyte/second', 'mebibyte', 'megahertz', 'microsecond', 'millisecond', 'millisecond/command', 'occurrence', 'operation', 'operation/second', 'packet', 'percent', 'request', 'second', 'watt']` none of which we expect to have negative values.

Also see page 167 of https://vdc-download.vmware.com/vmwb-repository/dcr-public/1b6cf07d-adb3-4dba-8c47-9c1c92b04857/241956dd-e128-4fcc-8131-bf66e1edd895/vcloud_sp_api_guide_30_0.pdf

Also this is something that is being done by other monitoring solutions for vSphere: https://github.com/influxdata/telegraf/blob/master/plugins/inputs/vsphere/endpoint.go#L985-L987

Reply:
One other possibility would be to only ignore values that are equals to `-1`, because this is the only value that I've seen so far. But it's hard to confirm that `-1` is the only value that means that the metric is not ready yet

Reply:
OK sounds good !